### PR TITLE
Add Fault to nova.ServerDetail for details of a nova instance in an E…

### PR DESCRIPTION
**Pull request by hmlanigan**
…RROR state.

Update testservices to allow for 'No valid host' error on specified availability zone.  ServerDetails will now return with server StatusActive, unless StatusError triggered with AZ.

Also tested against a live openstack where one network caused a 'No Valid Host' error.

### CODE REVIEW

File: testservices/novaservice/service_test.go [hunk 1]
```diff
@@ -252,29 +252,6 @@ func (s *NovaSuite) TestAddServerTwiceFails(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, `conflictingRequest: A server with id "test" already exists`)
 }
 
-// A control point can be used to change the status of the added server.
-func (s *NovaSuite) TestAddServerControlPoint(c *gc.C) {
```
Comment (babbageclunk) [inline testservices/novaservice/service_test.go]: Why was this deleted? Add the reason in the description if it was a driveby tidyup.

File: testservices/novaservice/service_test.go [hunk 2]
```diff
@@ -417,6 +397,7 @@ func (s *NovaSuite) TestAllServersWithMultipleFilters(c *gc.C) {
 	for _, server := range servers {
 		s.createServer(c, server)
 		defer s.deleteServer(c, server)
+		//server.Status = nova.StatusActive
```
Comment (babbageclunk) [inline testservices/novaservice/service_test.go]: Deliberate? Worth an explanatory comment if so.

**Review by babbageclunk — APPROVED** [review]
Looks good, a couple of questions again.

File: testservices/novaservice/service_test.go (hunk 2 above)
Comment (hmlanigan) [reply testservices/novaservice/service_test.go]: nope, removed.

File: testservices/novaservice/service_test.go (hunk 1 above)
Comment (hmlanigan) [reply testservices/novaservice/service_test.go]: Updated the test to change the server name via the hook.  Changing the server status will 
no longer work due to changes to the test service to mimic the behavior of server status
of building, active and error.

**Comment by hmlanigan** [comment]
$$merge$$
